library/dd-sdk-android-session-replay/src/main/kotlin/com/datadog/android/sessionreplay/internal/async/RecordedDataQueueHandler.kt

```diff
@@ -0,0 +1,177 @@
+            }
+        } catch (e: RejectedExecutionException) {
+            // TODO: REPLAY-1364 Add logs here once the sdkLogger is added
+        } catch (e: NullPointerException) {
+            // in theory will never happen but we'll log it
+            // TODO: REPLAY-1364 Add logs here once the sdkLogger is added
+        }
+    }
+
+    @WorkerThread
+    private fun triggerProcessingLoop(currentTime: Long) {
+        while (recordedDataQueue.isNotEmpty()) {
+            val nextItem = recordedDataQueue.peek()
+
+            if (nextItem != null) {
+                if (shouldRemoveItem(nextItem, currentTime)) {
+                    recordedDataQueue.poll()
+                } else if (nextItem.isReady()) {
+                    processItem()
+                } else {
+                    break
+                }
+            }
+        }
+    }
+
+    private fun processItem() {
+        when (val item = recordedDataQueue.poll()) {
+            is SnapshotRecordedDataQueueItem ->
+                processSnapshotEvent(item)
+            is TouchEventRecordedDataQueueItem ->
+                processTouchEvent(item)
+        }
+    }
+
+    private fun processSnapshotEvent(item: SnapshotRecordedDataQueueItem) {
+        processor.processScreenSnapshots(
+            newContext = item.newRumContext,
+            prevContext = item.prevRumContext,
+            timestamp = item.timestamp,
+            nodes = item.nodes,
+            systemInformation = item.systemInformation
+        )
+    }
+
+    private fun processTouchEvent(item: TouchEventRecordedDataQueueItem) {
+        processor.processTouchEventsRecords(
+            newContext = item.newRumContext,
+            touchEventsRecords = item.touchData
+        )
+    }
+
+    private fun shouldRemoveItem(recordedDataQueueItem: RecordedDataQueueItem, currentTime: Long) =
+        !recordedDataQueueItem.isValid() || isTooOld(currentTime, recordedDataQueueItem)
+
+    private fun isTooOld(currentTime: Long, recordedDataQueueItem: RecordedDataQueueItem): Boolean =
+        (currentTime - recordedDataQueueItem.timestamp) > MAX_DELAY_MS
+
+    private fun insertIntoRecordedDataQueue(recordedDataQueueItem: RecordedDataQueueItem) {
+        @Suppress("SwallowedException", "TooGenericExceptionCaught")
+        try {
+            recordedDataQueue.offer(recordedDataQueueItem)
+        } catch (e: IllegalArgumentException) {
+            // TODO: REPLAY-1364 Add logs here once the sdkLogger is added
+        } catch (e: ClassCastException) {
+            // in theory will never happen but we'll log it
+            // TODO: REPLAY-1364 Add logs here once the sdkLogger is added
+        } catch (e: NullPointerException) {
+            // in theory will never happen but we'll log it
+            // TODO: REPLAY-1364 Add logs here once the sdkLogger is added
+        }
+    }
+
+    // end region
+
+    internal companion object {
+        internal const val MAX_DELAY_MS = 200L
```

Comment:
are you sure you are able to load all the bitmaps in `200ms` ? I guess you will see once you add that logic and you can play with this number.

Reply:
Yes, this is just an arbitrary assumption. We'll need to test on an image-rich application to figure out what the threshold should be